File: angular-frontend/src/app/services/task/task.service.ts
import { Injectable } from '@angular/core';
import {Observable} from 'rxjs';
import {Task} from '../../models/task/task';
import {HttpClient} from '@angular/common/http';

const API_URL = 'http://localhost:8081/tasks';

@Injectable({
  providedIn: 'root'
})
export class TaskService {

  constructor(private httpClient: HttpClient) { }

  getAllByUser(id?: number): Observable<Task[]>{
    return this.httpClient.get<Task[]>(`${API_URL}/user/${id}`);
  }

  getAllByStudent(id?: number): Observable<Task[]>{
    return this.httpClient.get<Task[]>(`${API_URL}/student/${id}`);
  }

  create(task: Task): Observable<Task[]>{
    return this.httpClient.post<Task[]>(`${API_URL}`, task);
  }

  update(task: Task): Observable<Task[]>{
    return this.httpClient.put<Task[]>(`${API_URL}`, task);
  }

}

File: spring-security-backend/src/main/java/by/shilay/app/service/api/TaskService.java
package by.shilay.app.service.api;

import by.shilay.app.model.Task;

import java.util.List;

public interface TaskService {
    List<Task> getAll();
    List<Task> getAllByUser(Long id);
    List<Task> getAllByStudent(Long id);
    void create(Task task);
    void update(Task task);
}
